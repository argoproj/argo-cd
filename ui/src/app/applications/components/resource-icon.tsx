import * as React from 'react';
import {resources} from './resources';

export const ResourceIcon = ({kind, customStyle}: {kind: string; customStyle?: React.CSSProperties}) => {
    if (kind === 'node') {
        return <img src={'assets/images/infrastructure_components/' + kind + '.svg'} alt={kind} style={{padding: '2px', width: '40px', height: '32px', ...customStyle}} />;
    }
    const i = resources.get(kind);
    if (i !== undefined) {
        return <img src={'assets/images/resources/' + i + '.svg'} alt={kind} style={{padding: '2px', width: '40px', height: '32px', ...customStyle}} />;
    }
    if (kind === 'Application') {
        return <i title={kind} className={`icon argo-icon-application`} style={customStyle} />;
    }
    const initials = kind.replace(/[a-z]/g, '');
    const n = initials.length;
    const style: React.CSSProperties = {
        display: 'inline-block',
        verticalAlign: 'middle',
        padding: `${n <= 2 ? 2 : 0}px 4px`,
        width: '32px',
        height: '32px',
        borderRadius: '50%',
        backgroundColor: '#8FA4B1',
        textAlign: 'center',
        lineHeight: '30px',
        ...customStyle
    };
    return (
        <div style={style}>
            <span style={{color: 'white', fontSize: `${n <= 2 ? 1 : 0.6}em`}}>{initials}</span>
        </div>
    );
};
