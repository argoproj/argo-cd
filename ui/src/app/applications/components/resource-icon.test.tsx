import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {ResourceIcon} from './resource-icon';

test('ConfigMap', () => {
    const tree = renderer.create(<ResourceIcon kind='ConfigMap' />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('Application', () => {
    const tree = renderer.create(<ResourceIcon kind='Application' />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('OneWord', () => {
    const tree = renderer.create(<ResourceIcon kind='OneWord' />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('TwoWords', () => {
    const tree = renderer.create(<ResourceIcon kind='TwoWords' />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('ThreeWords', () => {
    const tree = renderer.create(<ResourceIcon kind='ThreeWordsFoo'/>).toJSON();

    expect(tree).toMatchSnapshot();
});

test('FourWords', () => {
    const tree = renderer.create(<ResourceIcon kind='FourWordsFooBar'/>).toJSON();

    expect(tree).toMatchSnapshot();
});
