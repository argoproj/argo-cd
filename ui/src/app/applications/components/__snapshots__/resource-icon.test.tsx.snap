// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application 1`] = `
<i
  className="icon argo-icon-application"
  title="Application"
/>
`;

exports[`ConfigMap 1`] = `
<img
  alt="ConfigMap"
  src="assets/images/resources/cm.svg"
  style={
    Object {
      "height": "32px",
      "padding": "2px",
      "width": "40px",
    }
  }
/>
`;

exports[`FourWords 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "height": "32px",
      "padding": "2px 4px",
      "width": "40px",
    }
  }
>
  <i
    className="icon fa fa-circle"
    title="FourWordsFooBar"
  />
  <div
    style={
      Object {
        "color": "white",
        "fontSize": "0.6em",
        "left": "10px",
        "margin": "auto",
        "position": "absolute",
        "textAlign": "center",
        "top": "14px",
        "width": "40px",
      }
    }
  >
    FWFB
  </div>
</div>
`;

exports[`OneWord 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "height": "32px",
      "padding": "2px 4px",
      "width": "40px",
    }
  }
>
  <i
    className="icon fa fa-circle"
    title="OneWord"
  />
  <div
    style={
      Object {
        "color": "white",
        "fontSize": "1em",
        "left": "10px",
        "margin": "auto",
        "position": "absolute",
        "textAlign": "center",
        "top": "10px",
        "width": "40px",
      }
    }
  >
    OW
  </div>
</div>
`;

exports[`ThreeWords 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "height": "32px",
      "padding": "2px 4px",
      "width": "40px",
    }
  }
>
  <i
    className="icon fa fa-circle"
    title="ThreeWordsFoo"
  />
  <div
    style={
      Object {
        "color": "white",
        "fontSize": "0.6em",
        "left": "10px",
        "margin": "auto",
        "position": "absolute",
        "textAlign": "center",
        "top": "14px",
        "width": "40px",
      }
    }
  >
    TWF
  </div>
</div>
`;

exports[`TwoWords 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "height": "32px",
      "padding": "2px 4px",
      "width": "40px",
    }
  }
>
  <i
    className="icon fa fa-circle"
    title="TwoWords"
  />
  <div
    style={
      Object {
        "color": "white",
        "fontSize": "1em",
        "left": "10px",
        "margin": "auto",
        "position": "absolute",
        "textAlign": "center",
        "top": "10px",
        "width": "40px",
      }
    }
  >
    TW
  </div>
</div>
`;
